src/components/dashboardCanvas/Canvas.jsx

```diff
@@ -8,58 +8,53 @@ export default class Canvas extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            divs: 1,
+            divs: [],
             moving: -1
         };
     }
     addListeners = () => {
-        console.log('here')
-        for (var index = 0; index < this.state.divs; index++) {
-            document.getElementById(`${index}`).addEventListener('mousedown', this.mousedown, false)
+        for (let index = 0; index < this.state.divs.length; index++) {
+            document.getElementById(`${index}header`).addEventListener('mousedown', this.mousedown, false)
             window.addEventListener('mouseup', this.mouseup, false)
         };
     }
     getDivs = () => {
-        var divs = [];
-        for (var ii = 0; ii < this.state.divs; ii++) {
+        let divs = [];
```

Comment:
If this is just for rendering purposes you don't need to set a state variable, you can just have a component function that returns the html you want to mount to the DOM. Its not necessarily an anti-pattern to have it as part of state, but in most cases its also not how we do it throughout the rest of the project